main.py

```diff
@@ -59,3 +73,55 @@ async def request_classification(request: Request):
             "classification_scores": json.dumps(classification_scores),
         },
     )
+
+
+@app.get("/histogram")
+# This is the get request for the histogram form
+async def get_histogram_form(request: Request):
+    return templates.TemplateResponse(
+        "histogram_select.html",
+        {
+            "request": request,
+            "images": list_images()
+        }
+    )
+
+
+@app.post("/histogram")
+# This is the post request for the histogram form
+async def request_histogram(request: Request):
+    form = HistogramForm(request)
+    await form.load_data()
+    # We retrieve the image id from the form
+    image_id = form.image_id
+
+    # We calculate the histogram scores for the image
+    histogram_scores = calculate_histogram(image_id)
+
+    # We save the histogram scores to the histogram_scores.txt file
+    with open('app/static/histogram_scores.txt', 'w') as f:
+        f.write(str(histogram_scores))
```

Comment:
What happens if multiple users ask for an image histogram? The file would be overwritten. Isn't it best to return the results directly? Or, if you reuse them later for other things, another way to do it would be to store the results with a unique name and clean up after the response is returned to the user (or after a while anyway).